src/ethereum/ethEngine.js

```diff
@@ -524,72 +531,66 @@ export class EthereumEngine extends CurrencyEngine {
     this.addTransaction(this.currencyInfo.currencyCode, edgeTransaction)
   }
 
-  // curreently for Ethereum but should allow other currencies
-  async checkUpdateNetworkFees() {
-    // Get the network fees from the info server
-    try {
-      const infoServer = getEdgeInfoServer()
-      const url = `${infoServer}/v1/networkFees/${this.currencyInfo.currencyCode}`
-      const jsonObj: EthereumFees = await this.ethNetwork.fetchGet(url)
-      const valid = asMaybe(asEthereumFees)(jsonObj) != null
-
-      if (valid) {
-        if (
-          JSON.stringify(this.walletLocalData.otherData.networkFees) !==
-          JSON.stringify(jsonObj)
-        ) {
-          this.walletLocalData.otherData.networkFees = jsonObj
-          this.walletLocalDataDirty = true
-        }
-      } else {
-        this.error(
-          `Error: Fetched invalid networkFees ${JSON.stringify(jsonObj)}`
-        )
+  /**
+   *  Fetch network fees from various providers in order of priority, stopping
+   *  and writing upon successful result.
+   */
+  async updateNetworkFees() {
+    // Get base fees from 'rpcServers' and convert to our network fees format.
+    // * Supported for ETH post EIP-1559 fork only
+    const isEth = this.currencyInfo.currencyCode === 'ETH'
```

Comment:
I've added a flag 'supportEIP1559' to the 'currencyInfo.defaultSettings.otherSettings' of the currencies you've mentioned and used that flag inside the `updateNetworkFeesFromBaseFeePerGas` function instead of checking it in the caller.